hs_core/views/__init__.py

```diff
@@ -527,6 +527,7 @@ def set_resource_flag(request, shortkey, *args, **kwargs):
     elif t == 'make_shareable':
        _set_resource_sharing_status(request, user, res, flag_to_set='shareable', flag_value=True)
 
+    request.session['resource-mode'] = request.POST.get('resource-mode', 'view')
```

Comment:
@pkdash If we want to stay in edit mode after users change resource flag, wouldn't it be sufficient to set ```resource-mode``` directly here to ```edit``` rather than retrieving it from request.POST data?